climate/plugins/oshosts/host_plugin.py

```diff
@@ -183,18 +190,25 @@ def update_reservation(self, reservation_id, values, usage_enforcement=False, us
         hosts_in_pool = pool.get_computehosts(
             reservation['resource_id'])
 
+        host_allocations = db_api.host_allocation_get_all_by_values(reservation_id=reservation_id)
+
+
         # Check if we have enough available SUs for update
         if usage_enforcement:
+            total_su_factor = sum(
```

Comment:
The update_reservation function can change the set of reserved hosts, so there is a possibility that for leases without node_type (created with the CLI), the newly allocated hosts may have a different su_factor. We should sum up the total_su_factor after the new matching hosts have been selected, but before the host allocations have been updated.